src/interface/hosted-buttons.js

```diff
@@ -0,0 +1,29 @@
+/* @flow */
+
+import { getHostedButtonsComponent } from "../hosted-buttons";
+import type { HostedButtonsComponent } from "../hosted-buttons/types";
+import { getButtonsComponent } from "../zoid/buttons";
+import {
+  getCardFormComponent,
+  type CardFormComponent,
+} from "../zoid/card-form";
+import { getCheckoutComponent, type CheckoutComponent } from "../zoid/checkout";
+import type { LazyExport, LazyProtectedExport } from "../types";
+import { protectedExport } from "../lib";
+
+export const HostedButtons: LazyExport<HostedButtonsComponent> = {
+  __get__: () => getHostedButtonsComponent(),
+};
+
+export const Checkout: LazyProtectedExport<CheckoutComponent> = {
+  __get__: () => protectedExport(getCheckoutComponent()),
+};
+
+export const CardForm: LazyProtectedExport<CardFormComponent> = {
+  __get__: () => protectedExport(getCardFormComponent()),
+};
+
+export function setup() {
+  getButtonsComponent();
+  getCheckoutComponent();
+}
```

Comment:
What does `setup` do?

Reply:
I'm not sure other than without it, `window.xprops` is undefined - and this line throws: https://github.com/paypal/paypal-smart-payment-buttons/blob/c268be3d0b18ff0ecaa11df646e74c8787ec03ff/src/button/button.js#L130

will dm an internal link to where `import {setup}` is being called

Reply:
oh I can say this generally:

in csnw: `/server/sdk/bootstrap` line # 35

Reply:
oh we do have this in `interface/buttons.js` this is fine